libs/config/value.go

```diff
@@ -0,0 +1,109 @@
+package config
+
+import "time"
+
+type Value struct {
+	v any
+	l Location
+
+	// Whether or not this value is an anchor.
+	// If this node doesn't map to a type, we don't need to warn about it.
+	anchor bool
+}
+
+// NilValue is equal to the zero-value of Value.
+var NilValue = Value{}
+
+// NewValue constructs a new Value with the given value and location.
+func NewValue(v any, loc Location) Value {
+	return Value{
+		v: v,
+		l: loc,
+	}
+}
+
+func (v Value) AsMap() (map[string]Value, bool) {
+	m, ok := v.v.(map[string]Value)
+	return m, ok
+}
+
+func (v Value) Location() Location {
+	return v.l
+}
+
+func (v Value) AsAny() any {
+	switch vv := v.v.(type) {
+	case map[string]Value:
+		m := make(map[string]any)
+		for k, v := range vv {
+			m[k] = v.AsAny()
+		}
+		return m
+	case []Value:
+		a := make([]any, len(vv))
+		for i, v := range vv {
+			a[i] = v.AsAny()
+		}
+		return a
+	case string:
+		return vv
+	case bool:
+		return vv
+	case int:
+		return vv
+	case int32:
+		return vv
+	case int64:
+		return vv
+	case float32:
+		return vv
+	case float64:
+		return vv
+	case time.Time:
+		return vv
+	case nil:
+		return nil
+	default:
+		panic("not handled")
+	}
```

Comment:
Good point -- I'd like to only deal with known types in this function. In a follow up PR (to be posted), I'm adding `config.Kind` field (similar to `reflect.Kind`) so we can capture the type of field here and do a switch/case on that. Then we can error on construction instead of erroring on `AsAny`.